### PR TITLE
Fix losses and aggregators when using CUDA graphs

### PR DESCRIPTION
**Pull request by jasooney23**
# PhysicsNeMo Pull Request

## Description

closes https://github.com/NVIDIA/physicsnemo-sym/issues/279

Passes `step` as an int scalar tensor to gradient computation functions so that CUDA graphs can track it properly.
Refactors loss functions and aggregators to replace `if` statements with `torch.where`s for CUDA graph compatibility.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo-sym/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo-sym/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/physicsnemo-sym/issues) is linked to this pull request.

### CODE REVIEW

**Review by loliverhennigh — APPROVED** [review]
I went throught the larger code blocks fairly carefully and seems correct to me. I would say if the unit tests run and we were able to do some sanity check on the examples we are good to go
